nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/download/DownloadPathParser.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * (C) Copyright 2016 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Estelle Giuly <egiuly@nuxeo.com>
+ */
+
+package org.nuxeo.ecm.core.io.download;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * This class allows to parse the path of a URL used to download blobs.
+ * It builds a map made of {repository, docId, xpath, filename}.
+ * For instance, it parses the download path "default/3727ef6b-cf8c-4f27-ab2c-79de0171a2c8/files:files/0/file/image.png"
+ * into { "repository": "default",
+ *        "docId": "3727ef6b-cf8c-4f27-ab2c-79de0171a2c8",
+ *        "xpath": "files:files/0/file",
+ *        "filename": "image.png" }
+ *
+ * @since 9.1
+ */
+public class DownloadPathParser {
```

Comment:
The users of the class just see it as a "download path", the parsing is a side-effect of what the constructor does. So I'd call this class just something like `DownloadBlobInfo`